### PR TITLE
Various doc updates and R CMD check fixes

### PR DESCRIPTION
**Pull request by leeper**
I made a bunch of fixes to the package that will help it to pass R CMD check. There are still a lot of undocumented parts of the package which are throwing warnings, but this fixes some of the import-related problems. I also added some documentation (converted the README to a knitr README.Rmd, added a CITATION file, and cleaned up the DESCRIPTION file to comply with CRAN policies).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,64 +11,98 @@ for quick analysis and plotting.
...
+
+```r
 patient_ages = fda_query("/drug/event.json") %>%
                fda_api_key("MYKEY") %>%
                fda_count("patient.patientonsetage") %>%
                fda_exec()
 ```
 
+```
+## Fetching: https://api.fda.gov/drug/event.json?search=&api_key=MYKEY&count=patient.patientonsetage
```
Comment (rjpower) [inline README.md]: Can we just drop the output for these entries (it's not super interesting) :)

File: README.md [hunk 2]
```diff
@@ -11,64 +11,98 @@ for quick analysis and plotting.
...
-You can filter the results to count on using the fda_filter() method:
+![plot of chunk unnamed-chunk-3](http://i.imgur.com/7qLI1Pe.png) 
 
-```
+You can filter the results to count on using the `fda_filter()` method:
+
+
+```r
 paxil_ages = fda_query("/drug/event.json") %>%
                fda_filter("patient.drug.openfda.generic_name", "paroxetine")
```
Comment (rjpower) [inline README.md]: Thanks: it looks like there's a missing %>% for this line that this conversion caught.

**Comment by rjpower** [comment]
This is awesome: thanks!  There we just a few nits otherwise it looks great.

**Comment by leeper** [comment]
@rjpower Cleaned up the changes. I can merge, but I thought I'd let you take a look first.

**Comment by rjpower** [comment]
LGTM.  Thanks again.

**Pull request merged**